### PR TITLE
Improve error logging in common test code

### PR DESCRIPTION
**Pull request by jaredmoo**
Write additional error details (including script stack trace) into test failure message in `EnvironmentSetupHelper.RunPowerShellTest`, and also write those details in test log error stream (`PowerShellExtensions.LogPowerShellResults`).

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/Common/Commands.Common/PowerShellUtilities.cs [hunk 1]
```diff
@@ -93,5 +93,16 @@ public static IEnumerable<RuntimeDefinedParameter> GetUsedDynamicParameters(Runt
         {
             return dynamicParameters.Values.Where(dp => MyInvocation.BoundParameters.Keys.Any(bp => bp.Equals(dp.Name)));
         }
+
+        public static string FormatErrorRecord(ErrorRecord record)
```
Comment (shahabhijeet) [inline src/Common/Commands.Common/PowerShellUtilities.cs]: @jaredmoo any reason a simple format wrapper function has to be introduced in the product.
if you need it for test reason, declare it locally inside EnvironmentSetupHelper.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs [hunk 2]
```diff
@@ -369,7 +369,12 @@ public virtual Collection<PSObject> RunPowerShellTest(params string[] scripts)
                     if (powershell.Streams.Error.Count > 0)
                     {
                         throw new RuntimeException(
-                            "Test failed due to a non-empty error stream, check the error stream in the test log for more details.");
+                            string.Format(
+                                "Test failed due to a non-empty error stream. First error: {0}{1}",
+                                PowerShellUtilities.FormatErrorRecord(powershell.Streams.Error[0]),
```
Comment (shahabhijeet) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs]: @jaredmoo either create a new local function in this file or simply inline the format wrapper function here.
It's little odd to call a function inside string.format() that basically calls string.format()
But if you feel strong to have a separate utility function for string.format, please declare locally where you have found the need to call it.
Else please let me know the scenarios where you will need the same function to be called from anywhere else.

Comment (jaredmoo) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs]: It's used in the next file https://github.com/Azure/azure-powershell/pull/3663/files#diff-90463d182d91894d58358adc66b14651R119

File: src/Common/Commands.Common/PowerShellUtilities.cs (hunk 1 above)
Comment (jaredmoo) [inline src/Common/Commands.Common/PowerShellUtilities.cs]: Yes, I'm extracting the code from https://github.com/Azure/azure-powershell/pull/3663/files#diff-90463d182d91894d58358adc66b14651L117 as a method. This file seemed like the logical place to put it.

**Comment by jaredmoo** [comment]
@shahabhijeet please see comments, I don't believe any changes are needed.

**Comment by shahabhijeet** [comment]
@jaredmoo please do not add wrapper function to the product. Plus we are working on two tasks that will improve error reporting.
If you feel strongly about this function, please either create a new wrapper or inline it inside EnvironmentSetupHelper.cs

**Pull request merged**